### PR TITLE
Release guide

### PR DESCRIPTION
**Pull request by LysandreJik**
This PR adds a small release guide to the `setup.py` for releases.

I added it here as it's the same location as for the `transformers` library and it seems adequate, but happy to move it somewhere else more appropriate.

### CODE REVIEW

**Comment by LysandreJik** [comment]
Following the guide to do a patch, I realize there's some info missing. Will complete once I'm done with the patch.

File: setup.py
Comment (adrinjalali) [inline setup.py:24]: do we want to maintain conda here? `transformers` doesn't seem to be uploading to our channel anymore, and we have our package on conda-forge. That recipe is more up-to-date.

File: setup.py
Comment (adrinjalali) [inline setup.py:38]: to give visibility to others, this can be a PR to the release branch instead of direct commit.

Comment (adrinjalali) [reply setup.py:38]: It would also help with checking the doc build on that PR.

File: setup.py
Comment (adrinjalali) [inline setup.py:46]: this can also be a PR

**Review by adrinjalali — COMMENTED** [review]
Thanks for the PR @LysandreJik . I think it'd be nice to move this to `doc/dev/release.mdx` rather than having it in `setup.py`.

File: setup.py
Comment (LysandreJik) [reply setup.py:24]: Right now the build works well and is automatic; I'd like to draft a small message to share with the rest of the team regarding conda support and whether we want to continue to support it through our own channel or not. In the meantime, I'd like to keep it here.

File: setup.py
Comment (LysandreJik) [reply setup.py]: How would you name the branch from which that PR is made? Others will get visibility with the release already, I am not 100% sure I see why the PR is necessary here

Comment (adrinjalali) [reply setup.py]: that PR comes from the maintainer's own fork, doesn't matter how it's named. E.g: https://github.com/scikit-learn/scikit-learn/pull/21120

In most cases it'll be a simple PR, but sometimes you might want to include more commits from the `main` branch.

**Review by adrinjalali — APPROVED** [review]
This is a good start anyway. I'm happy for this to be merged, updated or not :)

**Comment by LysandreJik** [comment]
I'm having a hard time finding coherent names for the branches as I haven't followed that workflow previously; I'll merge this as it is, but very open if you want to open another PR to complete the existing doc with the two additional branches :)

**Pull request merged**
